pkg/scaffold/v2/controller/controller_suitetest.go

```diff
@@ -32,16 +33,25 @@ var _ input.File = &ControllerSuiteTest{}
 type ControllerSuiteTest struct {
 	input.Input
 
-	// Resource is the resource to scaffold the controller_kind_test.go file for
+	// Resource is the Resource to make the Controller for
 	Resource *resource.Resource
 }
 
 // GetInput implements input.File
 func (v *ControllerSuiteTest) GetInput() (input.Input, error) {
+
 	if v.Path == "" {
-		v.Path = filepath.Join("controllers", "suite_test.go")
+		if v.MultiGroup {
+			v.Path = filepath.Join("controllers",
+				v.Resource.Group,
+				"suite_test.go")
+		} else {
+			v.Path = filepath.Join("controllers", "suite_test.go")
+		}
 	}
+
 	v.TemplateBody = controllerSuiteTestTemplate
+	v.Input.IfExistsAction = input.Error
```

Comment:
I'm a bit worried about this, it seems to work but my logic sais it shouldn't.

Reply:
To add a note:

Just to ensure it we can see the testdata where all is gen with success. 

<img width="493" alt="Screenshot 2020-01-07 at 15 19 30" src="https://user-images.githubusercontent.com/7708031/71907248-8caea300-3163-11ea-825c-c9d47d9be112.png">

The same behaviour done for a single group is now done per group.

Reply:
@Adirio what specifically are you concerned about here?

Reply:
Lets say Admiral, Captain and Firstmate are created in that order. If I understand that flag correctly, all the three will try to scaffold the suite_test.go file. Admiral will not find the file so no problem. But then Captain does see the file when he tries to create it, as it was already created by Admiral. And if the flag is set to Error when the file exists, my logic sais that it should fail. But as we can see in the testdata is not failing. So it may be me missunderstanding that flag or something. I expected it to be set to input.Skip in order not to do anythin if the suite test file was already created.

Reply:
I still not getting the concern with since this PR does not change the logic to create/update the suit_tests. The same behaviour done when it has just 1 group is what we do for each new one.

Reply:
What @Adirio said seems to be reasonable. 
If that's the case, I wonder why the golden test doesn't catch this :thinking: 

This line probably should be moved into `if v.MultiGroup {}` statement.

Reply:
I am still unable to check why with multigroup would an issue be faced with that cannot be faced with the single group layout. However, moved the line to the if as suggested.
Thank you for the review and collab.

Reply:
> What @Adirio said seems to be reasonable.
> If that's the case, I wonder why the golden test doesn't catch this 🤔

I'm really trying to discover why it is not behaving the way I expected. There is a switch that returns an error if this flag is set to `input.Error` as expected (`pkg/scaffold/scaffold.go:259-269`) and this function is called in a loop but any error will make it stop (`pkg/scaffold/scaffold.go:210-214`). Can it be related to the way we are detecting if a file is already present (`pkg/scaffold/scaffold.go:178-183`)? I suspect I have already found some problems with this in an unrelated PR when dealing with the `testdata` directory.

> This line probably should be moved into `if v.MultiGroup {}` statement.

Shouldn't it just be removed? I mean there are still several controllers potentially sharing the suite test file, why should it give an error?

Reply:
Hi @mengqiy,
 
IHMO we should not change the logic here and we should keep it as it has been working in the master. See that in the master we also are able to add many kinds of controllers for the same suit_test. I mean the same scenario occurs with the single group layout. Make no sense for me we change the code base and/or try to find a solution for a problem that we are unable to face/see and or reproduce.

Reply:
@camilamacedo86 Exactly, master was not adding this flag. This PR added it. Thats why I think that you should not add it even if it seems to working by some strange magic we are unable to understand.